### DISCUSSION TITLE
New to ducted aircons and controllers etc

### DISCUSSION THREAD

**Discussion by Danielson89** [omerfaruk-aran/esphome_samsung_hvac_bus#222] [General]
Hi,
We recently purchased a house that came with a Samsung Ducted A/C, according to the paperwork it was installed in 2019. You can purchase the Samsung Wi-Fi kit for $499AUD. I recently built a wireless controller using an ESP8266 for another A/C in the same house. It was a Midea split system A/C, and that works very well with Home Assistant.  

I was wondering will this Samsung_HVAC project work with ducted or even mine?  I don't have the model number and need to get it from the outdoor unit.

Thanks

**Comment by omerfaruk-aran** [comment]
Hi,

Congratulations on your new home! Yes, the Samsung_HVAC project may work with your ducted Samsung A/C system, but it will depend on the specific model and communication protocols of your unit. Once you get the model number from the outdoor unit, we can verify if it’s compatible with this integration.

If it was installed in 2019, there’s a good chance it supports the necessary RS485 communication, which is what the Samsung_HVAC project uses to interface with Home Assistant.

Feel free to share the model number once you have it, and I’ll be happy to guide you through the setup process if it’s compatible.

Thanks!

**Reply by Danielson89** [reply]
Thank you :)

When I get the model number tonight after work,  Lets say it's compatible,  does the ESP device need to be connected on the box inside the roof space that I believe that is where the motor unit is.

Thanks

**Reply by tech2urdoor** [reply]
Hi Daniel,

Whatever you do don't buy the wifi kit from Samsung! I have a Samsung ducted ac and the wifi kit. MIM-H02 and they stopped supporting the app that I used to control it and pulled the app from the playstore. I could have purchased an upgraded Wifi-kit for $380AUD but they would have inevitably just pulled support for that too. I can't speak for your model but I was able to disconnect the wifi kit and attach the ESP to the same wires that was hooked up to. I believe it is wired into the control panel next to it but I can't be 100% sure as it was all installed when I moved into the house. I didn't have to remove the controller. I'll attach a photo of it all. The ESP fitted nicely inside the old wifi kit so you can't see it.

![PXL_20241112_053002294](https://github.com/user-attachments/assets/ca44af47-214e-45be-a8f6-e9ff9648159c)

**Reply by Danielson89** [reply]
OOOOOOOO I've got that exact same controller for the air con.  Was the wires for the wifi controller 4 of them?

**Reply by Danielson89** [reply]
> Hi,
> 
> Congratulations on your new home! Yes, the Samsung_HVAC project may work with your ducted Samsung A/C system, but it will depend on the specific model and communication protocols of your unit. Once you get the model number from the outdoor unit, we can verify if it’s compatible with this integration.
> 
> If it was installed in 2019, there’s a good chance it supports the necessary RS485 communication, which is what the Samsung_HVAC project uses to interface with Home Assistant.
> 
> Feel free to share the model number once you have it, and I’ll be happy to guide you through the setup process if it’s compatible.
> 
> Thanks!

So I got all the parts required,  I ran into an issue my ESP8266 said cannot upload due to low disk space?  I assume that’s why somewhere I read use ESP32?

Last night I tried using an Arduino Uno R4 Wifi and Raspberry Pi Zero W2 but got stuck so I abandoned it.

**Comment by tech2urdoor** [comment]
I didn't have to open the controller. I was lucky that I only had to open the wifi kit and disconnect those wires from the wifi kit and attach them to the ESP. Yes there was four. Here is a picture of the back of the wifi kit and the wires going into it
![PXL_20241022_053804651](https://github.com/user-attachments/assets/48c3af64-580e-41bf-ae08-65fdc83b2ef9)

**Comment by tech2urdoor** [comment]
I did have to change some programming within the controller to tell it that there was a secondary controller (wifi kit). This was never done as the previous owners obviously never actually used the wifi kit and the installers didn't configure the controller for it. I'll try and dig up the site I referred to when changing that setting...it was quite a while ago...

**Comment by tech2urdoor** [comment]
https://forums.whirlpool.net.au/thread/98mz6263?p=2
https://eu.community.samsung.com/t5/smart-home/smart-home-app-not-working/m-p/1388322#M210

There is a bit of talk here about it all. I know I had to change the setting in my controller to make the wifi kit function. You might have to do the same with yours although I'm not 100% sure if you are going straight to the ESP.
![Screenshot_20200701-224517_Samsung%20Members](https://github.com/user-attachments/assets/dfb0a836-2b15-4ba5-a4ed-5e2403f69b0f)

**Reply by Danielson89** [reply]
Yep will be using an ESP8266 as I have bulk of them

**Reply by Danielson89** [reply]
> https://forums.whirlpool.net.au/thread/98mz6263?p=2 https://eu.community.samsung.com/t5/smart-home/smart-home-app-not-working/m-p/1388322#M210
> 
> There is a bit of talk here about it all. I know I had to change the setting in my controller to make the wifi kit function. You might have to do the same with yours although I'm not 100% sure if you are going straight to the ESP. ![Screenshot_20200701-224517_Samsung%20Members](https://private-user-images.githubusercontent.com/57508933/385210217-dfb0a836-2b15-4ba5-a4ed-5e2403f69b0f.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzIyNDUxMDQsIm5iZiI6MTczMjI0NDgwNCwicGF0aCI6Ii81NzUwODkzMy8zODUyMTAyMTctZGZiMGE4MzYtMmIxNS00YmE1LWE0ZWQtNWUyNDAzZjY5YjBmLmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTIyVDAzMDY0NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFlMmVhODE0YjVlNDczZTE3YzdmNmRmMzJmY2Q1NjU0YTIxODhhYTExZGNkZTM4ZTIzOTdhY2FmZTVkY2M1ZjAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Pb3ch08BB7YwjLQsjnh_A-VXdVXZKZNcBjR4JlsfNUI)

unfortunately my air condition isn't supported with the wifi module,  I'm believing the 2019 models must of not had that,  I'm assuming the 2018 and below supported wifi?  -  I cannot seem to get any luck from mine.

**Comment by Danielson89** [comment]
Sorry images didn't load take 2

Please see the attached photos of my model and controller board,  it has bulk connections

**Comment by Danielson89** [comment]
![IMG_9901](https://github.com/user-attachments/assets/b35ea7bc-889f-4a50-bc23-4ac5b7eb09b9)
![IMG_9899](https://github.com/user-attachments/assets/1905097b-692f-4f49-b04d-2be0ab015a42)
![IMG_9898](https://github.com/user-attachments/assets/
![IMG_9897](https://githu
![IMG_9896](https://github.com/user-attachments/assets/3f334567-79cc-47bb-b076-4cecfd476be8)
b.com/user-attachments/assets/308deed5-f6c9-443
![IMG_9893](https://github.com/user-attachments/assets/ae114e94-cc10-42c6-81fa-d0a6ff8398ec)
![IMG_9897](https://github.com/user-attachments/assets/f8538664-449a-4f37-aa6d-77aa97691d4a)

![IMG_9902](https://github.com/user-attachments/assets/a9c06fcf-0935-4ce2-b55f-49d960565acd)

**Reply by tech2urdoor** [reply]
![PXL_20241001_070647925](https://github.com/user-attachments/assets/0ba6f167-e516-467b-9d42-e851fdd171c3)

Here is a pic of my outdoor unit. It looks like your unit is just a more powerful version of mine.
If you are using an esp8266 do you have a rs485 adaptor for it? Are those photos from the inside of the outdoor unit? I just presumed you would be wiring it to the control panel like mine would be....but perhaps you can wire it into the outdoor unit.

**Reply by Danielson89** [reply]
Ahh yep,  One of the photos is the outdoor unit and the rest are the indoor unit inside the roof space in the house.  Ohhh do you just tap into the control panel?  Otherwise I'm happy to run the ESP8266 + RS485 inside the indoor unit and tuck it in to that space.  

I see that it's 12v on the diagram my ESP8266 is only a 5v version.  I also purchased a RS485 yesterday,  waiting delivery but got a pack of like 10 of the ESP8266's,  I've got heaps of Raspberry Pi 4's, 3's Model B's,  Got the newest Arduino with that small cluster of LED's on the board as well.  I wonder If I can use anything else?

Thanks

**Reply by tech2urdoor** [reply]
Ah gotcha. Don't mind me, I was just presuming my wires were coming from the control panel next to it. They probably are coming from the indoor unit in the roof. I can't really access my roof very easily. I was very lucky that I could just disconnect the wires from the wifi kit and wire them into the atom chip.

**Reply by mvisic** [reply]
> ![PXL_20241001_070647925](https://private-user-images.githubusercontent.com/57508933/385663977-0ba6f167-e516-467b-9d42-e851fdd171c3.jpg?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjI2MDg3MzAsIm5iZiI6MTc2MjYwODQzMCwicGF0aCI6Ii81NzUwODkzMy8zODU2NjM5NzctMGJhNmYxNjctZTUxNi00NjdiLTlkNDItZTg1MWZkZDE3MWMzLmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTExMDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMTA4VDEzMjcxMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWMyM2Q3ZmQ2OGYzMGEwZDg0MzYxNWI0NmQ0ZGFmY2I1NWY1Y2JiZDcyZTgyYmZlNzk1ZjM2MjAyOWYyMjBjYzQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.3rdHD-KagPoiQfWlqUe9pZqXNvuirJxF1vIclFTtkYo)
> 
> Here is a pic of my outdoor unit. It looks like your unit is just a more powerful version of mine. If you are using an esp8266 do you have a rs485 adaptor for it? Are those photos from the inside of the outdoor unit? I just presumed you would be wiring it to the control panel like mine would be....but perhaps you can wire it into the outdoor unit.

I have the exact same model outdoor unit and am looking forward to wiring up an ESP32 + rs485 device. How's it all going one year on? Any other advice?

**Reply by Danielson89** [reply]
> > ![PXL_20241001_070647925](https://private-user-images.githubusercontent.com/57508933/385663977-0ba6f167-e516-467b-9d42-e851fdd171c3.jpg?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjI2MDg3MzAsIm5iZiI6MTc2MjYwODQzMCwicGF0aCI6Ii81NzUwODkzMy8zODU2NjM5NzctMGJhNmYxNjctZTUxNi00NjdiLTlkNDItZTg1MWZkZDE3MWMzLmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTExMDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMTA4VDEzMjcxMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWMyM2Q3ZmQ2OGYzMGEwZDg0MzYxNWI0NmQ0ZGFmY2I1NWY1Y2JiZDcyZTgyYmZlNzk1ZjM2MjAyOWYyMjBjYzQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.3rdHD-KagPoiQfWlqUe9pZqXNvuirJxF1vIclFTtkYo)
> > Here is a pic of my outdoor unit. It looks like your unit is just a more powerful version of mine. If you are using an esp8266 do you have a rs485 adaptor for it? Are those photos from the inside of the outdoor unit? I just presumed you would be wiring it to the control panel like mine would be....but perhaps you can wire it into the outdoor unit.
> 
> I have the exact same model outdoor unit and am looking forward to wiring up an ESP32 + rs485 device. How's it all going one year on? Any other advice?

Personal recommendations,  Use M5Stacks for simplicity,  1 yearish on we use it every single day it's just sooo awesome!!!!,  I have heaps more advice when needed

**Reply by mvisic** [reply]
@Danielson89 I'm just waiting on the M5Stack to arrive. Did you end up wiring it into the indoor unit in the ceiling? If so, did you do it yourself or get a sparky? By the way, I'm also in Perth.

**Reply by Danielson89** [reply]
> @Danielson89 I'm just waiting on the M5Stack to arrive. Did you end up wiring it into the indoor unit in the ceiling? If so, did you do it yourself or get a sparky? By the way, I'm also in Perth.

I did it myself,  I probably should have turned the power off but I didn't.  It didn't break anything but I recommend power off!!

Got into the roof space,  mine had a cool inspection door with 2 screws,  V1 and V2 went directly into the 12vDC and GND of the M5 Stacks and F1, F2 to A and B and the code that "[tech2urdoor]" shared, and it just worked.

I had some 4 core alarm cable and used about 30cm for ease of access.  That is all you need the M5Stacks RS485,  Atom Lite and just plugged the Atom on to the M5Stacks body.   Sweet yeah I'm in Perth as well,  Just near Joondalup.

**Comment by Danielson89** [comment]
Ok so I've discovered that F1 and F2 are the outdoor unit and it doesn't like me hooking up the TX RX wires to that as it lost comms and I've put them on F3 and F4,  had to change the wiring a little bit because the indoor controller lost comms so swapped the TX and RX around and comms works again.  I have absolutely no idea.

**Reply by Danielson89** [reply]
@tech2urdoor @omerfaruk-aran - any more assistance you might have?

**Reply by tech2urdoor** [reply]
All I can suggest is buying the atom stack that's recommended, changing the settings in your control panel to tell it there is a slave controller and wiring it up to the F1 and F2 terminals. This would make it the same setup as mine which should work since we have almost identical models.

**Reply by Danielson89** [reply]
> All I can suggest is buying the atom stack that's recommended, changing the settings in your control panel to tell it there is a slave controller and wiring it up to the F1 and F2 terminals. This would make it the same setup as mine which should work since we have almost identical models.

Thanks - Is that all ok if my Outdoor unit is F1 and F2 terminals?

**Reply by Danielson89** [reply]
> All I can suggest is buying the atom stack that's recommended, changing the settings in your control panel to tell it there is a slave controller and wiring it up to the F1 and F2 terminals. This would make it the same setup as mine which should work since we have almost identical models.

Ok so the M5stacks finally arrived I've hooked it up to F1 and F2 terminals which connects to the compressor outside but still not working.  I feel like I've got a unique unit that was the only 1 that Samsung made that didn't support any aftermarket wifi controllers.  I'm going to purchase the genuine Samsung AP set up as I think we know works.  I wasted 40 Australian dollars on that M5stacks.

**Comment by Danielson89** [comment]
Removed message,  got very frustrated,  I should of slept on it first :)

**Reply by tech2urdoor** [reply]
Sorry for the late reply. I haven't really been following this project as I sort of just set and forget it. As I said, I just replaced my genuine samsung ap with this and it worked perfectly. If you went with a genuine one and somehow managed to get that wired up correctly then you can guarantee the M5Stack setup would work as well. I don't suppose you are in SA? If you are close enough to me i'd happily take a look at it for you or you could come and see how mine is wired up.

**Reply by Danielson89** [reply]
> Sorry for the late reply. I haven't really been following this project as I sort of just set and forget it. As I said, I just replaced my genuine samsung ap with this and it worked perfectly. If you went with a genuine one and somehow managed to get that wired up correctly then you can guarantee the M5Stack setup would work as well. I don't suppose you are in SA? If you are close enough to me i'd happily take a look at it for you or you could come and see how mine is wired up.

Thanks,  sorry I got very frustrated last night because I was getting reading from the outdoor unit and it was annoying me because I thought after 6 months of trying, I thought I was there....  But nope.  South Africa or South Australia?   If any of those two sorry I'm in Perth WA.

Out of thought here is the script I'm using.  I'm using the M5Stacks RS485 with the M5Stacks Atom mounted on top of it.

Just to clear things up.  I tried wiring up directly to my wired remote on the wired remote I had the following pins

V1 and V2 = DC12v and GND
F3 & F4 
On the indoor unit within the roof space, I can see F1 and F2,   V1 and V2 and F3 and F4

Currently, it's wired up to 
F1 = A,  
F2 = B. 

and V1 is plugged directly into 12v on the 12v port of the M5Stacks RS485,  V2 to the GND of the M5Stacks




Here is the code in using in ESPHome

esphome:
  name: samsungac
  friendly_name: Climate

esp32:
  board: m5stack-atom

web_server:
  port: 80

logger:
  level: VERY_VERBOSE

api:

ota:
  platform: esphome

wifi:
  ssid: "Removed"
  password: "Removed"

captive_portal:

uart:
  tx_pin: GPIO19
  rx_pin: GPIO22
  baud_rate: 9600
  parity: EVEN

external_components:
  - source: github://omerfaruk-aran/esphome_samsung_hvac_bus@main
    components: [samsung_ac]

samsung_ac:
  non_nasa_keepalive: true
  debug_log_messages: true
  debug_log_undefined_messages: true

  capabilities:
    vertical_swing: true
    horizontal_swing: true

    presets:
      quiet: true
      fast:
        name: "Fast cooling"
        enabled: true

  devices:
    - address: "00.00.00"
      climate:
        name: "Kitchen climate"
      room_temperature:
        name: "Kitchen temperature"
      target_temperature:
        name: "Kitchen target temperature"
      power:
        name: "Kitchen power"
      mode:
        name: "Kitchen mode"
      automatic_cleaning:
        name: "Kitchen automatic clean"
      room_temperature_offset: -1.4
      room_humidity:
        name: "Kitchen humidity"

    - address: "C8.00.00"
      error_code:
        name: "Outdoor Error Code"
      outdoor_instantaneous_power:
        name: "Outdoor Instantaneous Power"
      outdoor_cumulative_energy:
        name: "Outdoor Cumulative Energy"
      outdoor_current:
        name: "Outdoor Current"
      outdoor_voltage:
        name: "Outdoor Voltage"
      outdoor_temperature:
        name: "Outdoor Temperature"

https://github.com/user-attachments/assets/3a50c7b7-544d-47cb-8e65-b911259ea390


![image](https://github.com/user-attachments/assets/db1cdf2e-d982-4d71-a377-aa33a1011a62)

**Reply by tech2urdoor** [reply]
I think your problem might be that your address isn't just "00" and "c8" like mine is? you have .00.00 at the end and in the instructions I think that's only for NASA devices and yours is Non-NASA like mine? Maybe...I'm not too sure. Something to try anyway. I'm so sorry for the late reply once again. Personal issues. I understand you would be frustrated, I would be too so don't apologise. I'll try to watch this thread a bit closer. Definitely try just having the two digit addresses and let me know.

**Reply by tech2urdoor** [reply]
[atom_yaml.txt](https://github.com/user-attachments/files/21334207/atom_yaml.txt)

**Reply by Danielson89** [reply]
> [atom_yaml.txt](https://github.com/user-attachments/files/21334207/atom_yaml.txt)
> 
> 

Wait is that all you have in your yaml?? Omg wow ok,  I've tried that and have completely failed,  It must be my wiring then?

Do you remember how you hooked your one up?  As in what connections etc?? 

I've tried F1/F2, F3/F4 swapped around tried many many many different codes codes but nothing.

My current wiring is the M5Stacks RS485 and Atom lite bolted on top,  RS485 is getting power from V1/V2 as it's 12v RS485 and then F1 and F2 in the respective A and B ports

Oh my gawd I cannot wait to solve it. 

Thanks

**Reply by tech2urdoor** [reply]
Yep that was the code that worked for me. Very basic just following the example. As for wiring I just used the wires that were going to the existing wifi controller so I couldn't tell you where they actually went but I just used the V1,V2 to power it and then the F1, F2 to A and B. It sounds like your wiring is correct. Have you made sure that the setting in your wired controller is setup for a secondary controller like I spoke about earlier in this thread. I remember it's something I had to do just to get the genuine wifi controller actually talking to it. Obviously the previous owners that installed it never even used the wifi capabilities as they wouldn't have been able to as the controller wasn't programmed for it.

**Reply by Danielson89** [reply]
> Yep that was the code that worked for me. Very basic just following the example. As for wiring I just used the wires that were going to the existing wifi controller so I couldn't tell you where they actually went but I just used the V1,V2 to power it and then the F1, F2 to A and B. It sounds like your wiring is correct. Have you made sure that the setting in your wired controller is setup for a secondary controller like I spoke about earlier in this thread. I remember it's something I had to do just to get the genuine wifi controller actually talking to it. Obviously the previous owners that installed it never even used the wifi capabilities as they wouldn't have been able to as the controller wasn't programmed for it.

OMG OMG OMG IT WORKED!!!!!!!!!!!!!!!! AHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHH

ALL I DID was change the wifi address from you script,  left everything else as is and pressed install and instantly it just started working AHHHHHHHHHHHHHHHHHHHHHHH 8 MONTHS of trying

**Reply by tech2urdoor** [reply]
Yay! I'm happy you finally got it working mate. What a relief that must be for you. I'm sorry I didn't share my yaml with you earlier, its just so stock standard I didn't think it would be of any use.

**Comment by Jkirjo** [comment]
Is really indoor address only zeroes? I don't know, but think it is not usually so.

**Reply by Danielson89** [reply]
Great question.  I could only get 00 from "Indoor" and outdoor is c8

**Reply by tech2urdoor** [reply]
<img width="3798" height="1722" alt="image" src="https://github.com/user-attachments/assets/1801c6c7-6ce5-48d4-8710-ec5e72688708" />

**Comment by homepersonalprojects** [comment]
@tech2urdoor - Can you please, please, please share your code?? I have the same unit and I'm simply unable to flash my ESP32. The code compiles fine but when I flash and try to check on logs, I can see that the bootloader is being reset. I've done multiple tests to ensure that the hardware is not bad. Would really appreciate your code to be shared here.

**Reply by tech2urdoor** [reply]
I'm not sure how it happened but at some stage it disappeared from the esp home builder in HA but it is still accessible in home assistant. So I can operate it but I can't edit or even see the code. I posted my code here at some stage though when I was trying to get accurate power readings. I would have to pull open the old wifi controller box that I managed to squeeze the m5 unit into. Then I would have to track down the code I put in here myself at some stage. Unless someone knows how to get something back into the builder? I have many esp devices in there but for some reason that is the only device that has done that....

**Reply by tech2urdoor** [reply]
I found my post. This would have been my working yaml : 
[atom_yaml.txt](https://github.com/user-attachments/files/21334142/atom_yaml.txt)

I followed the directions and I can't say I had to change anything out of the ordinary. It all just worked but I did have the benefit of the existing samsung wifi kit so I never had to worry about where the wiring was going.

**Reply by homepersonalprojects** [reply]
Wow! Didn't expect a reply and ended up troubleshooting for a good 3 hours. I've got my own code and included LED status indicators to it so that I know what's happening in the future. I still haven't been able to test my code yet but will probably be trying over the next couple of days. Will have to experiment with both your code and mine.

**Comment by VladCocan** [comment]
> Hi, We recently purchased a house that came with a Samsung Ducted A/C, according to the paperwork it was installed in 2019. You can purchase the Samsung Wi-Fi kit for $499AUD. I recently built a wireless controller using an ESP8266 for another A/C in the same house. It was a Midea split system A/C, and that works very well with Home Assistant.
> 
> I was wondering will this Samsung_HVAC project work with ducted or even mine? I don't have the model number and need to get it from the outdoor unit.
> 
> Thanks

All Samsung units sold after 2014, are using the same protocol, it dosn`t mater what form factor it is (ducted unit, wall mount, 4way, 360,etc...), the only difference is about optional accessories and functions (horizontal/vertical louvre, windfree,, SPI, etc...), the basic functions (mode, fan speed, setpoint)are the same for all units, this apply to RAC (AR*), CAC (AC*), EHS(AE*), TDM(AE*), FJM(AJ*), DVM(AM*), and in case you have multiple outdoor units, you can daisy chain all units on port R1/R2, and use only one ESP32.

**Reply by Danielson89** [reply]
> > Hi, We recently purchased a house that came with a Samsung Ducted A/C, according to the paperwork it was installed in 2019. You can purchase the Samsung Wi-Fi kit for $499AUD. I recently built a wireless controller using an ESP8266 for another A/C in the same house. It was a Midea split system A/C, and that works very well with Home Assistant.
> > I was wondering will this Samsung_HVAC project work with ducted or even mine? I don't have the model number and need to get it from the outdoor unit.
> > Thanks
> 
> All Samsung units sold after 2014, are using the same protocol, it dosn`t mater what form factor it is (ducted unit, wall mount, 4way, 360,etc...), the only difference is about optional accessories and functions (horizontal/vertical louvre, windfree,, SPI, etc...), the basic functions (mode, fan speed, setpoint)are the same for all units, this apply to RAC (AR*), CAC (AC*), EHS(AE*), TDM(AE*), FJM(AJ*), DVM(AM*), and in case you have multiple outdoor units, you can daisy chain all units on port R1/R2, and use only one ESP32.

Thanks Since my original message,  I've managed to get it all going and working correctly with Home Assistant.  I ended up using the M5Stacks RS485 and the Atom Lite ESP32.  

I've also redone my zone controls,  I used a 8 way relay board and a Pi zero and now my zones are automated,  instead of the push button switches.

**Reply by tech2urdoor** [reply]
Hey Daniel,

I don't suppose you can share with me the process of you automating your zones. I would love to be able to set zones with mine. At the moment I just have a dumb zonemaster controller with four capacitive buttons on it. I would love to be able to control that somehow remotely. I just haven't figured out the best way yet. Cheers.

**Reply by Danielson89** [reply]
> Hey Daniel,
> 
> 
> 
> I don't suppose you can share with me the process of you automating your zones. I would love to be able to set zones with mine. At the moment I just have a dumb zonemaster controller with four capacitive buttons on it. I would love to be able to control that somehow remotely. I just haven't figured out the best way yet. Cheers.

Yes,  absolutely.  I'm just on my way to work.  I'll share it when I get there.  I'll have to say first it isn't the best way but I didn't have an option because my wall controller broke.  It works and it's based on an MQTT script I made up.  The process is 8way Relay board from AliExpress.  Relay 1 = Zone 1 On,  Relay 2 = Zone 1 Off.  The power was shared between each.  The AC connected to Common on the relay and the other end to the grounds of the zone contollers.  I know you can get the iZone system but I wasn't prepared to spend $1400.  

Equipment needed for this setup
Raspberry Pi Zero W (Nice and small)
8way relay board
Home Assistant 
a total cost of $35
